examples/ng_6lo_udp/main.c

```diff
@@ -0,0 +1,189 @@
+    if (ip == NULL) {
+        puts("Error: unable to allocate IPv6 header");
+        ng_pktbuf_release(udp);
+        return;
+    }
+    /* send packet */
+    sendto = ng_netreg_lookup(NG_NETTYPE_UDP, NG_NETREG_DEMUX_CTX_ALL);
+    if (sendto == NULL) {
+        puts("Error: unable to locate UDP thread");
+        ng_pktbuf_release(ip);
+        return;
+    }
+    ng_pktbuf_hold(ip, ng_netreg_num(NG_NETTYPE_UDP,
+                                     NG_NETREG_DEMUX_CTX_ALL) - 1);
+    while (sendto != NULL) {
+        ng_netapi_send(sendto->pid, ip);
+        sendto = ng_netreg_getnext(sendto);
+    }
+    printf("Success: send %i byte to %s:%u\n", payload->size, addr_str, tmp);
+}
+
+static void start_server(char *port_str)
+{
+    uint16_t port;
+
+    /* check if server is already running */
+    if (server.pid != KERNEL_PID_UNDEF) {
+        printf("Error: server already running on port %" PRIu32 "\n",
+                server.demux_ctx);
+        return;
+    }
+    /* parse port */
+    port = (uint16_t)atoi(port_str);
+    if (port == 0) {
+        puts("Error: invalid port specified");
+        return;
+    }
+    /* start server (which means registering pktdump for the chosen port) */
+    server.pid = ng_pktdump_getpid();
+    server.demux_ctx = (uint32_t)port;
+    ng_netreg_register(NG_NETTYPE_UDP, &server);
+    printf("Success: started UDP server on port %" PRIu16 "\n", port);
+}
+
+static void stop_server(void)
+{
+    /* check if server is running at all */
+    if (server.pid == KERNEL_PID_UNDEF) {
+        printf("Error: server was not running\n");
+        return;
+    }
+    /* stop server */
+    ng_netreg_unregister(NG_NETTYPE_UDP, &server);
+    server.pid = KERNEL_PID_UNDEF;
+    puts("Success: stopped UDP server");
+}
+
+static int udp_cmd(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s [send|server]\n", argv[0]);
+        return 1;
+    }
+
+    if (strcmp(argv[1], "send") == 0) {
+        if (argc < 5) {
+            printf("usage: %s send <addr> <port> <data>\n", argv[0]);
+            return 1;
+        }
+        send(argv[2], argv[3], argv[4]);
+    }
+    else if (strcmp(argv[1], "server") == 0) {
+        if (argc < 3) {
+            printf("usage: %s server [start|stop]\n", argv[0]);
+            return 1;
+        }
+        if (strcmp(argv[2], "start") == 0) {
+            if (argc < 4) {
+                printf("usage %s server start <port>\n", argv[0]);
+                return 1;
+            }
+            start_server(argv[3]);
+        }
+        else if (strcmp(argv[2], "stop") == 0) {
+            stop_server();
+        }
+        else {
+            puts("error: invalid command");
+        }
+    }
+    else {
+        puts("error: invalid command");
+    }
+    return 0;
+}
+
+static const shell_command_t shell_commands[] = {
+    { "udp", "send data over UDP and listen on UDP ports", udp_cmd },
+    { NULL, NULL, NULL }
+};
+
```

Comment:
Maybe all those above function could have been written in a file named ``udp.c`` ?